src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebRequestPSCmdlet.CoreClr.cs

```diff
@@ -408,6 +412,34 @@ protected override void StopProcessing()
 
         #region Helper Methods
 
+        private void ThrowNotSuccessStatusCodeError(HttpRequestMessage request, HttpResponseMessage response)
+        {
+            var detailMsg = "";
+            var reader = new StreamReader(StreamHelper.GetResponseStream(response));
+            try
+            {
+                detailMsg = reader.ReadToEnd();
+            }
+            catch (IOException)
+            {
+            }
+            catch (ArgumentException)
+            {
+            }
+            finally
+            {
+                reader.Dispose();
+            }
+
+            var msg = string.Format(CultureInfo.CurrentCulture, WebCmdletStrings.NotSuccessStatusCode, (int) response.StatusCode, response.ReasonPhrase);
+            ErrorRecord er = new ErrorRecord(new WebException(msg) { Response = response }, "WebCmdletWebResponseException", ErrorCategory.InvalidOperation, request);
```

Comment:
`Exception.Data` was a neat idea, but it's not discoverable and a poor api. Because you typically derive from `System.Exception`, it's easier to add your data as properties - and more reliable - your property can be readonly.